projects/pgai/tests/vectorizer/test_vectorizer_cli.py

```diff
@@ -88,39 +87,37 @@ def cli_db_url(cli_db: tuple[TestDatabase, Connection]) -> str:
 
 def test_worker_no_tasks(cli_db_url: str):
     """Test that worker handles no tasks gracefully"""
-    result = CliRunner().invoke(vectorizer_worker, ["--db-url", cli_db_url, "--once"])
+    result = run_vectorizer_worker(cli_db_url)
 
     # It exits successfully
     assert result.exit_code == 0
     assert "no vectorizers found" in result.output.lower()
 
 
-@pytest.fixture
-def source_table(
-    cli_db: tuple[TestDatabase, Connection], test_params: tuple[int, int, int, str, str]
-) -> str:
-    _, conn = cli_db
-    num_items = test_params[0]
+def setup_source_table(
+    connection: Connection,
+    number_of_rows: int,
+):
     table_name = "blog"
-    with conn.cursor(row_factory=dict_row) as cur:
+    with connection.cursor(row_factory=dict_row) as cur:
         # Create source table
         cur.execute(f"""
-            CREATE TABLE {table_name} (
-                id INT NOT NULL PRIMARY KEY,
-                id2 INT NOT NULL,
-                content TEXT NOT NULL
-            )
-        """)
+                CREATE TABLE {table_name} (
+                    id INT NOT NULL PRIMARY KEY,
+                    id2 INT NOT NULL,
+                    content TEXT NOT NULL
+                )
+            """)
         # Insert test data
-        values = [(i, i, f"post_{i}") for i in range(1, num_items + 1)]
+        values = [(i, i, f"post_{i}") for i in range(1, number_of_rows + 1)]
         cur.executemany(
             "INSERT INTO blog (id, id2, content) VALUES (%s, %s, %s)", values
         )
     return table_name
 
 
 @pytest.fixture(scope="session")
-def ollama_connection_url():
+def ollama_container_url_or_url():
```

Comment:
This confuses me, what about just `ollama_url`?

Reply:
yeh fair, I was just confused why this is a fixture and not a constant 😅 which is why i put the container in there, but tbh fair to encapsulate and hide the details.